src/software/ai/hl/stp/evaluation/possession_test.cpp

```diff
@@ -190,22 +190,30 @@ TEST(PossessionEvaluationTest, team_had_possession_half_second_ago)
 
     ball.updateState(
         {0.1, 2.5}, {0, 10},
-        Timestamp::fromSeconds(2 + Util::DynamicParameters::Evaluation::Possession::
-                                       possession_buffer_time_seconds.value()));
+        Timestamp::fromSeconds(2 + Util::DynamicParameters->getEvaluationConfig()
+                                       ->getPossessionConfig()
+                                       ->PossessionBufferTimeSeconds()
+                                       ->value()));
     world.updateBallState(ball);
 
     robot0.updateState(
         Point(-2, 3), Vector(), Angle::zero(), AngularVelocity::zero(),
-        Timestamp::fromSeconds(2 + Util::DynamicParameters::Evaluation::Possession::
-                                       possession_buffer_time_seconds.value()));
+        Timestamp::fromSeconds(2 + Util::DynamicParameters->getEvaluationConfig()
+                                       ->getPossessionConfig()
+                                       ->PossessionBufferTimeSeconds()
+                                       ->value()));
     robot1.updateState(
         Point(-2, 0), Vector(), Angle::quarter(), AngularVelocity::zero(),
-        Timestamp::fromSeconds(2 + Util::DynamicParameters::Evaluation::Possession::
-                                       possession_buffer_time_seconds.value()));
+        Timestamp::fromSeconds(2 + Util::DynamicParameters->getEvaluationConfig()
+                                       ->getPossessionConfig()
+                                       ->PossessionBufferTimeSeconds()
+                                       ->value()));
```

Comment:
Like Gareth mentioned, this would be much nicer if we stored `Util::DynamicParameters->getEvaluationConfig()->getPossessionConfig()->PossessionBufferTimeSeconds()->value()));` once at the top of this function and reused that value

Reply:
I agree, but this is a lot of work to do across all the files, to a point to where you might as well do dependency injection at the same time if we are going through the trouble of splitting it up.

I think this file is especially bad so I can do this one, but across everything is a lot of work to do as part of this PR

Reply:
I agree, out of scope, file an issue

Reply:
There is this ticket for dependency injection #960 which I think will address this by nature of splitting things up

Reply:
Cool